src/main/java/org/snomed/snowstorm/validation/ConceptValidationThenOperationService.java

```diff
@@ -0,0 +1,223 @@
+		CREATE,
+		UPDATE
+	}
+
+	@Autowired
+	private ConceptService conceptService;
+
+	@Autowired
+	private DroolsValidationService validationService;
+
+	@Autowired
+	private ObjectMapper objectMapper;
+
+	private ConceptValidationOperation operation;
+	private Concept concept;
+	private String acceptLanguageHeader;
+	private String branchPath;
+
+	public final ConceptValidationThenOperationService withParameters(final Concept concept, final String acceptLanguageHeader, final String branchPath) {
+		this.concept = generateUUIDSIfNotSet(concept);
+		this.acceptLanguageHeader = acceptLanguageHeader;
+		this.branchPath = branchPath;
+		return this;
+	}
+
+	private Concept generateUUIDSIfNotSet(final Concept concept) {
+		if (concept != null) {
+			if (concept.getConceptId() == null) {
+				concept.setConceptId(generateRandomUUID());
+			}
+			concept.getDescriptions().stream().filter(description -> description != null && description.getId() == null)
+					.forEach(description -> description.setDescriptionId(generateRandomUUID()));
+			concept.getRelationships().stream().filter(relationship -> relationship != null && relationship.getId() == null)
+					.forEach(relationship -> relationship.setRelationshipId(generateRandomUUID()));
+			concept.getAllOwlAxiomMembers().stream().filter(referenceSetMember -> referenceSetMember != null && referenceSetMember.getId() == null)
+					.forEach(referenceSetMember -> referenceSetMember.setRefsetId(generateRandomUUID()));
+		}
+		return concept;
+	}
+
+	/**
+	 * Generates a random UUID with length == 10 characters.
+	 *
+	 * @return random UUID with character length of 10.
+	 */
+	public static String generateRandomUUID() {
+		final int power = (int) Math.pow(10, (double) 10 - 1);
+		return "" + power + new Random().nextInt(9 * power);
+	}
+
+	public final ConceptValidationThenOperationService thenDo(final ConceptValidationOperation operation) {
+		this.operation = operation;
+		return this;
+	}
+
+	public final ResponseEntity<ConceptView> execute() throws JsonProcessingException, ServiceException {
+		throwExceptionIfFieldsNotSet();
+		final List<InvalidContent> invalidContents = validationService.validateConcept(branchPath, concept);
+		final List<InvalidContent> invalidContentErrors = invalidContents.stream().filter(invalidContent -> invalidContent.getSeverity() == Severity.ERROR).collect(Collectors.toList());
+		if (invalidContentErrors.isEmpty()) {
+			return operation == ConceptValidationOperation.CREATE ? createConcept(concept, acceptLanguageHeader, branchPath, invalidContents.stream().filter(invalidContent ->
+					invalidContent.getSeverity() == Severity.WARNING).collect(Collectors.toList())) : updateConcept(concept, acceptLanguageHeader, branchPath,
+					invalidContents.stream().filter(invalidContent -> invalidContent.getSeverity() == Severity.WARNING).collect(Collectors.toList()));
+		}
+		throw new DroolsValidationException(objectMapper.writeValueAsString(invalidContentErrors).replace("\n", ""));
```

Comment:
Nice catch replacing the newlines. If this replacement was made while setting the header it would be clearer why it's happening, otherwise a comment would be good.